jobqueue.go

```diff
@@ -166,7 +241,38 @@ func (s *Stack) Do(job func()) error {
 	return nil
 }
 
+// Status returns snapshot information about the state of the queue.
+func (s *Stack) Status() Status {
+	req := make(chan Status)
+	select {
+	case <-s.hasQuit:
+		return Status{Closed: true}
+	case s.status <- req:
+		return <-req
+	}
+
+}
+
 // Close frees up the resources used by a Stack instance.
+//
+// After called, the queue stops accepting new jobs, but it waits until all the
+// jobs all done, including those waiting in the queue.
+//
+// If the close timeout is set to >0, then forces closing after the timeout
```

Comment:
Where is the timeout handled?

Reply:
It's passed in as an option to the queue. I'll add this in the code doc. It's handled in the control loop.